rfcs/XXXX-separate-wal.md

```diff
@@ -0,0 +1,113 @@
+# Separate Object Store for WAL
+
+Status: Draft
+
+Authors:
+
+* [Sergei Sitnikov](https://github.com/taburet)
+
+## Motivation
+
+Sequential writes are a common pattern. Combined with `await_durable` set to
+`true`, which is the default, WAL performance is the main limiting factor for
+the sequential writes throughput. Even if `flush_interval` is set to an extremely
+low value, it can't overcome the inherent latency of the underlying object store.
+
+The obvious solution is switching to a lower latency object store; for instance,
+from S3 to S3 Express One Zone. But lower latency stores usually come with a cost,
+they might be significantly more expensive or might be limited in features. Using
+such stores for the main store containing the entire database might be undesirable.
+
+## Goals
+
+- Support an optional separate object store dedicated specifically for WAL.
+- Provide a user-facing configuration mechanism.
+- Avoid significant refactorings.
+
+## Public API
+
+The only change on the public API level is configuration:
+
+```rust
+pub struct DbOptions {
+    // ...
+
+    /// An optional [object store](ObjectStore) dedicated specifically for WAL.
+    ///
+    /// If not set, the main object store passed to `Db::open(...)` will be used
+    /// for WAL storage.
+    ///
+    /// NOTE: WAL durability properties depend on the durability properties of
```

Comment:
Very minor nit, but I think we should highlight availability as well, here. "durability and availability properties of..."

Reply:
https://github.com/slatedb/slatedb/pull/533/commits/21960e7041a343a16ef559ad30904242a6789380